Tutorial - How to only record the sessions you want (#5613)
PostHog/posthog.com: contents/tutorials/limit-session-recordings.md

---
title: How to only record the sessions you want
date: 2023-03-22
author: ["ian-vanagas"]
showTitle: true
sidebar: Docs
featuredImage: ../images/tutorials/banners/tutorial-17.png
topics: ["session recordings", "configuration"]
---

> Session recordings help you get a deep understanding of how users are using your product. They are our top recommendation for [early-stage startups](/blog/early-stage-analytics), but as you scale the number of recordings can go beyond what you need.

Reviewer: ```suggestion
Session recordings help you get a deep understanding of how users are using your product. We strongly recommend them for [early-stage startups](/blog/early-stage-analytics), but as you scale the number of recordings can go beyond what you need.
```

> Instead of turning session recordings off entirely, you can use PostHog’s configuration options to only record the sessions you want. This tutorial shows you how.

Reviewer: ```suggestion
Instead of turning session recordings off entirely, you can use PostHog’s configuration options to only record the sessions you want. This tutorial shows you three ways to do this. 
```

## Configuration and basic session recording controls

> The following ways to limit the number of sessions recorded require that `disable_session_recording` is set to `true` in the initialization of PostHog. For example, in a Next.js app, this looks like this:

Reviewer: ```suggestion
In order to follow this tutorial, you need to set `disable_session_recording` to `true` in PostHog's initialization. For example, in a Next.js app, this looks like this:
```

```js
// pages/_app.js
import posthog from "posthog-js"
import { PostHogProvider } from 'posthog-js/react'

if (typeof window !== 'undefined') {
  posthog.init('<ph_project_api_key>', {
    api_host: '<ph_instance_address>',
    disable_session_recording: true,
  })
}

export default function App({ Component, pageProps }) {
  return (
    <>
      <PostHogProvider client={posthog}>
        <Component {...pageProps} />
      </PostHogProvider>
    </>
  )
}
```

You can then use PostHog’s `startSessionRecording()` method to trigger a session recording at the right moment. As a basic example, we can use the initialization `loaded` method to start a session recording when the environment isn’t set to `development`.

```js
//...
if (typeof window !== 'undefined') {
  posthog.init('<ph_project_api_key>', {
    api_host: '<ph_instance_address>',
    disable_session_recording: true,
    loaded: (posthog) => {
      if (process.env.NODE_ENV != 'development') posthog.startSessionRecording()
    }
  })
}
//...
```

## Control recordings with feature flags

> You can use feature flags to control what sessions you record. This enables you to control the recording of sessions with: 

Reviewer: ```suggestion
Alternatively, you can use feature flags to control what sessions you record. This enables you to control the recording of sessions based on: 
```

- people properties
- group properties
- cohorts
- percentages of each of these (or all users)

> For example, if you only wanted to record 50% of North American users’ sessions, set up a feature flag where the continent code equals "NA" and roll it out to 50% of matching users like this:

Reviewer: ```suggestion
For example, if you only want to record 50% of sessions for North American users, set up a feature flag where the continent code equals "NA" and roll it out to 50% of matching users like this:
```

![Flag](../images/tutorials/limit-session-recordings/flag.png)

You can then check this flag after you initialize PostHog, and start a recording if it is active for the current user. To ensure the correct user and flag details, you can use the `usePostHog` and `useFeatureFlagEnabled` hooks from `posthog-js/react` in a component like this:

```js
import { useFeatureFlagEnabled, usePostHog } from 'posthog-js/react'

export default function Home() {
  const posthog = usePostHog()
  
  if (useFeatureFlagEnabled('record-na')) {
    posthog.startSessionRecording()
  }

  return (
    <>
      <p>Record this page</p>
    </>
  )
}
```

## Record on specific pages

> You can start recordings on specific pages by calling `startSessionRecording()` when those pages first load.

Reviewer: ```suggestion
You can gather recordings on specific pages by calling `startSessionRecording()` when those pages first load.
```

Author: It's more start than gather, gather implies that it is only that page.
